homeassistant/components/ondilo_ico/sensor.py

```diff
@@ -51,12 +50,19 @@ async def async_setup_entry(hass, entry, async_add_entities):
 
     api = hass.data[DOMAIN][entry.entry_id]
 
-    def get_all_pool_data(pool):
-        """Add pool details and last measures to pool data."""
-        pool["ICO"] = api.get_ICO_details(pool["id"])
-        pool["sensors"] = api.get_last_pool_measures(pool["id"])
+    def get_all_pools_data(api):
```

Comment:
We can move this function out to the module level.

Reply:
Thanks for the feedback, I see what you mean. In this case, I think it's cleaner to put that call at the api class level. I just pushed that change. I hope that's fine for you.